conans/test/integration/build_requires/test_build_requires_source_method.py

```diff
@@ -0,0 +1,91 @@
+import textwrap
+
+from conans.test.utils.tools import TestClient
+
+
+def test_build_requires_source():
+    c = TestClient()
+    tool = textwrap.dedent(r"""
+        import os
+        from conan import ConanFile
+        from conan.tools.files import chdir, save
+
+        class Tool(ConanFile):
+            name = "tool"
+            version = "0.1"
+            def package(self):
+                with chdir(self, self.package_folder):
+                    echo = f"@echo off\necho MY-TOOL! {self.name}/{self.version}!!"
+                    save(self, "bin/mytool.bat", echo)
+                    save(self, "bin/mytool.sh", echo)
+                    os.chmod("bin/mytool.sh", 0o777)
+        """)
+    pkg = textwrap.dedent("""
+        from conan import ConanFile
+        from conan.tools.env import buildenv
+        import platform
+
+        class Pkg(ConanFile):
+            name = "pkg"
+            version = "0.1"
+            tool_requires = "tool/0.1"
+
+            def source(self):
+                cmd = "mytool.bat" if platform.system() == "Windows" else "mytool.sh"
+                with buildenv(self):
+                    self.run(cmd)
```

Comment:
I have tried this approach, as we discussed, but this doesn't seem possible.
The test fail at the local ``conan source`` command, while trying to execute this, because it doesn't have enough information to construct the ``buildenv``, the ``conan source`` command do not construct the graph or have dependencies at all. Inhibiting this so it doesn't do anything and it is no-op when it is a local ``conan source`` command is not straightforward at all.

Reply:
This was my fear when first proposed. Do you have any preferred fallback for this case?